specfile/sourcelist.py

```diff
@@ -0,0 +1,120 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import collections
+from typing import List, Optional, overload
+
+from typing_extensions import SupportsIndex
+
+from specfile.rpm import Macros
+from specfile.sections import Section
+from specfile.tags import Comments
+
+
+class SourcelistEntry:
+    """
+    Class that represents a spec file source/patch.
```

Comment:
```suggestion
    Class that represents a spec file source/patch in a sourcelist/patchlist.
```

Reply:
Thanks. I'm thinking about making one small change - to make the attributes of `TagSource` and `ListSource` private, to make it clear they are not to be accessed by end-user. Do you think it makes sense?

Reply:
Yes, that makes sense IMO. This API is on a higher level and provides all information about the sources/patches that the users may need so it does not make sense to allow access to the lower level tag. Moreover, it could result in weird behavior if the underlying tag is modified so it's probably better to prevent it. Or can you see a use-case where we would need to access the lower level tag directly?

Reply:
No, I can't. Ok, I'm going to make the change and then I'll merge this.